vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,322 @@
+		}
+
+		replicas := 1
+		if item.Spec.Replicas != nil {
+			replicas = int(*item.Spec.Replicas)
+		}
+
+		item.Annotations[constants.PausedAnnotation] = "true"
+		item.Annotations[constants.PausedReplicasAnnotation] = strconv.Itoa(replicas)
+		item.Spec.Replicas = &zero
+
+		patch := client.MergeFrom(originalObject)
+		data, err := patch.Data(&item)
+		if err != nil {
+			return false, errors.Wrap(err, "create statefulSet patch")
+		}
+
+		// patch deployment
+		log.Infof("Scale down statefulSet %s/%s...", namespace, item.Name)
+		_, err = kubeClient.AppsV1().StatefulSets(namespace).Patch(context.TODO(), item.Name, patch.Type(), data, metav1.PatchOptions{})
+		if err != nil {
+			return false, errors.Wrap(err, "patch statefulSet")
+		}
+
+		// wait until deployment is scaled down
+		err = wait.PollImmediate(time.Second, time.Minute*3, func() (done bool, err error) {
+			obj, err := kubeClient.AppsV1().StatefulSets(namespace).Get(context.TODO(), item.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+
+			return obj.Status.Replicas == 0, nil
+		})
+		if err != nil {
+			return false, errors.Wrap(err, "wait for statefulSet scaled down")
+		}
+	}
+
+	return true, nil
+}
+
+// Resume
+
+func ResumeVCluster(kubeClient *kubernetes.Clientset, name, namespace string, log log.Logger) error {
+	// scale down vcluster itself
+	labelSelector := "app=vcluster,release=" + name
+	found, err := scaleUpStatefulSet(kubeClient, labelSelector, namespace, log)
+	if err != nil {
+		return err
+	} else if !found {
+		found, err = scaleUpDeployment(kubeClient, labelSelector, namespace, log)
+		if err != nil {
+			return err
+		} else if !found {
+			return errors.Errorf("couldn't find a paused vcluster %s in namespace %s. Make sure the vcluster exists and was paused previously", name, namespace)
+		}
+
+		// scale down kube api server
+		_, err = scaleUpDeployment(kubeClient, "app=vcluster-api,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+
+		// scale down kube controller
+		_, err = scaleUpDeployment(kubeClient, "app=vcluster-controller,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+
+		// scale down etcd
+		_, err = scaleUpStatefulSet(kubeClient, "app=vcluster-etcd,release="+name, namespace, log)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func scaleUpDeployment(kubeClient kubernetes.Interface, labelSelector string, namespace string, log log.Logger) (bool, error) {
+	list, err := kubeClient.AppsV1().Deployments(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return false, err
+	} else if len(list.Items) == 0 {
+		return false, nil
+	}
+
+	for _, item := range list.Items {
+		if item.Annotations == nil || item.Annotations[constants.PausedAnnotation] != "true" {
+			return false, nil
+		}
+
+		originalObject := item.DeepCopy()
+
+		replicas := 1
+		if item.Annotations[constants.PausedReplicasAnnotation] != "" {
```

Comment:
If the replica was a different number before pause, shouldn't we bring up the same count?

Reply:
Yes! If you look a few lines down you'll see that the old replica count is retrieved from an annotation. 1 is just the default.